### PR TITLE
Added print statements for clarity in provider yaml checks

### LINKED ISSUE #17316: Docs validation function - Add meaningful errors
The following function just prints right/left error which is not very meaningful and very difficult to troubleshoot.

```python
def check_doc_files(yaml_files: Dict[str, Dict]):
    print("Checking doc files")
    current_doc_urls = []
    current_logo_urls = []
    for provider in yaml_files.values():
        if 'integrations' in provider:
            current_doc_urls.extend(
                guide
                for guides in provider['integrations']
                if 'how-to-guide' in guides
                for guide in guides['how-to-guide']
            )
            current_logo_urls.extend(
                integration['logo'] for integration in provider['integrations'] if 'logo' in integration
            )
        if 'transfers' in provider:
            current_doc_urls.extend(
                op['how-to-guide'] for op in provider['transfers'] if 'how-to-guide' in op
            )
    expected_doc_urls = {
        "/docs/" + os.path.relpath(f, start=DOCS_DIR)
        for f in glob(f"{DOCS_DIR}/apache-airflow-providers-*/operators/**/*.rst", recursive=True)
        if not f.endswith("/index.rst") and '/_partials' not in f
    }
    expected_doc_urls |= {
        "/docs/" + os.path.relpath(f, start=DOCS_DIR)
        for f in glob(f"{DOCS_DIR}/apache-airflow-providers-*/operators.rst", recursive=True)
    }
    expected_logo_urls = {
        "/" + os.path.relpath(f, start=DOCS_DIR)
        for f in glob(f"{DOCS_DIR}/integration-logos/**/*", recursive=True)
        if os.path.isfile(f)
    }
    try:
        assert_sets_equal(set(expected_doc_urls), set(current_doc_urls))
        assert_sets_equal(set(expected_logo_urls), set(current_logo_urls))
    except AssertionError as ex:
        print(ex)
        sys.exit(1)
```

### PR DESCRIPTION
**Pull request by subkanthi**
closes: #17316 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
some static (ups!)  checks fail :D

File: scripts/ci/pre_commit/pre_commit_check_provider_yaml_files.py [hunk 1]
```diff
@@ -119,13 +119,13 @@ def assert_sets_equal(set1, set2):
 
     lines = []
     if difference1:
-        lines.append('Items in the first set but not the second:')
+        lines.append('    -- Items in the left set but not the right:')
         for item in sorted(difference1):
-            lines.append(repr(item))
+            lines.append('       ' + repr(item))
```
Comment (uranusjr) [inline scripts/ci/pre_commit/pre_commit_check_provider_yaml_files.py]:
```suggestion
            lines.append(f'       {item!r}')
```

File: scripts/ci/pre_commit/pre_commit_check_provider_yaml_files.py [hunk 2]
```diff
@@ -119,13 +119,13 @@ def assert_sets_equal(set1, set2):
...
     if difference2:
-        lines.append('Items in the second set but not the first:')
+        lines.append('    -- Items in the right set but not the left:')
         for item in sorted(difference2):
-            lines.append(repr(item))
+            lines.append('       ' + repr(item))
```
Comment (uranusjr) [inline scripts/ci/pre_commit/pre_commit_check_provider_yaml_files.py]:
```suggestion
            lines.append(f'       {item!r}')
```

**Review by uranusjr — CHANGES_REQUESTED** [review]

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
still some static test fail (also build images) - can you please use pre-commits and rebase to latest main ?

**Comment by subkanthi** [comment]
> still some static test fail (also build images) - can you please use pre-commits and rebase to latest main ?

Thanks @potiuk , fixed the static checks, but the 2 image failures doesnt seem to be related to the commits

**Comment by potiuk** [comment]
> Thanks @potiuk , fixed the static checks, but the 2 image failures doesnt seem to be related to the commits

Yeah. Those are unrelated.

**Review by potiuk — APPROVED** [review]

**Pull request merged**